src/connector/src/sink/google_pubsub.rs

```diff
@@ -0,0 +1,279 @@
+    #[serde(rename = "pubsub.project_id")]
+    pub project_id: String,
+
+    /// Specifies the Pub/Sub topic to publish messages
+    #[serde(rename = "pubsub.topic")]
+    pub topic: String,
+
+    /// The Google Pub/Sub endpoint URL
+    #[serde(rename = "pubsub.endpoint")]
+    pub endpoint: String,
+
+    /// use the connector with a pubsub emulator
+    /// <https://cloud.google.com/pubsub/docs/emulator>
+    #[serde(rename = "pubsub.emulator_host")]
+    pub emulator_host: Option<String>,
+
+    /// A JSON string containing the service account credentials for authorization,
+    /// see the [service-account](https://developers.google.com/workspace/guides/create-credentials#create_credentials_for_a_service_account) credentials guide.
+    /// The provided account credential must have the
+    /// `pubsub.publisher` [role](https://cloud.google.com/pubsub/docs/access-control#roles)
+    #[serde(rename = "pubsub.credentials")]
+    pub credentials: Option<String>,
+
+    // accept "append-only"
+    pub r#type: String,
+}
+
+impl GooglePubSubConfig {
+    fn from_hashmap(values: HashMap<String, String>) -> Result<Self> {
+        serde_json::from_value::<GooglePubSubConfig>(
+            serde_json::to_value(values).expect("impossible"),
+        )
+        .map_err(|e| SinkError::Config(anyhow!(e)))
+    }
+
+    pub(crate) fn initialize_env(&self) {
+        tracing::debug!("setting pubsub environment variables");
+        if let Some(emulator_host) = &self.emulator_host {
+            std::env::set_var("PUBSUB_EMULATOR_HOST", emulator_host);
+        }
+        if let Some(credentials) = &self.credentials {
+            std::env::set_var("GOOGLE_APPLICATION_CREDENTIALS_JSON", credentials);
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct GooglePubSubSink {
+    pub config: GooglePubSubConfig,
+    is_append_only: bool,
+
+    schema: Schema,
+    pk_indices: Vec<usize>,
+    format_desc: SinkFormatDesc,
+    db_name: String,
+    sink_from_name: String,
+}
+
+impl Sink for GooglePubSubSink {
+    type Coordinator = DummySinkCommitCoordinator;
+    type LogSinker = AsyncTruncateLogSinkerOf<GooglePubSubSinkWriter>;
+
+    const SINK_NAME: &'static str = PUBSUB_SINK;
+
+    fn is_sink_decouple(desc: &SinkDesc, user_specified: &SinkDecouple) -> Result<bool> {
+        match user_specified {
+            SinkDecouple::Default => Ok(desc.sink_type.is_append_only()),
+            SinkDecouple::Disable => Ok(false),
+            SinkDecouple::Enable => Ok(true),
+        }
+    }
+
+    async fn validate(&self) -> Result<()> {
+        if !self.is_append_only {
+            return Err(SinkError::Nats(anyhow!(
+                "Google Pub/Sub sink only support append-only mode"
+            )));
+        }
+        if !matches!(self.format_desc.encode, SinkEncode::Json) {
+            return Err(SinkError::GooglePubSub(anyhow!(
+                "Google Pub/Sub sink only support `Json` sink encode"
+            )));
+        }
```

Comment:
please impl this in `CONNECTORS_COMPATIBLE_FORMATS` (`src/frontend/src/handler/create_sink.rs`)